### DISCUSSION TITLE
Color and Dark mode glitches

### DISCUSSION THREAD

**Discussion by DaDuFan** [quasarframework/quasar#9767] [General - Components / Directives / etc] [answered]
Hi,
I noticed some minor color flaws.

1. In the default project template the Dark mode is not working correctly.
    The background of the drawer is always "bg-grey-1" (*)
    When you turn Dark mode on, the "Essential Links" are no longer readable, because the text color changes, but the background not (grey on grey)
    
    To reproduce it, simply create a new project

       ` quasar create qtest`
    and alway accept the defaults (or change it, it doesn't matter)

    edit "pages/Index.vue" and add
        `<q-btn flat round @click="$q.dark.toggle()" :icon="$q.dark.isActive ? 'nights_stay' : 'wb_sunny'" />`

    Start quasar ("`quasar dev`"), and click on the added button.
    
2. For me the Quasar-Logo is nearly unreadable in Dark Mode.
3. In the documentation e.g. https://quasar.dev/style/shadows I have real problems to read the light-blue text on white background

Anybode else having these problems?

(*) Possible solutions for 1. above

1. make the drawer class reactive and use an other class for Dark mode

2.  make an own css-class and use it in the drawer:
```
.body--light {
  .bg-my-drawer {
    background: #123456 !important;
  }
}
.body--dark {
  .bg-my-drawer {
    background: #654321 !important;
  }
}
```

3. like 2, but overwrite "bg-grey-1" in dark mode. This seems a bit harsh, bacause grey would be the new black ;-)
    But on the other side the text would be readable again.
    But when you wanted "bg-grey-1" to stay grey in Dark mode, is is a bad idea.
    
What do you think?
Any other option?
Which of the above ones is the best solution?
Atm I tend to use 2.

**Comment by rstoenescu** [comment, marked as answer]
The starter kit is not meant to be a full-blown app. It's just something very basic to get you started.

Obviously if you want to support Dark mode you'll need to adapt it to your needs. Which means removing the "bg-grey-1" on the drawer and applying dynamic classes or the body--light/body--dark selector. Same for any other area.

Telling Quasar to always use bg-grey-1 for the drawer and then expecting some other color on the screen is not right. Same for the transparent logo which is meant for light mode, not dark.

**Comment by sasgosh** [comment]
You can use solution 4:

First, you make "classes", like this:

```
colors: {
    button: () => {return $q.dark.isActive ? 'bg-light-blue-10 text-white' : 'bg-red-10 text-green'},
    drawer: () => {return $q.dark.isActive ? 'bg-deep-purple-10 text-red-6' : 'bg-deep-purple-1 text-green-10'},
    toolbar: () => {return $q.dark.isActive ? 'bg-green-10 text-yellow-1' : 'bg-deep-purple-1 text-gray-10'},
    ...
},
```

and then on elements:
```
<q-btn :class="colors.button()">
```

**Reply by pdanpdan** [reply]
Almost good, except it should be a computed and the condition should be only once

**Reply by sasgosh** [reply]
Yes, it can be done in many ways, for example:

```
codes: {
    button: ['bg-black text white', 'bg-white text-black'],
    drawer: ['bg-black text white', 'bg-white text-black'],
    toolbar: ['bg-black text white', 'bg-white text-black'],
}
color: (el) => {return $q.dark.isActive ? codes[el][0] : codes[el][1]}

```

```
<q-btn :class="color('button')">
```
